feat(core-kit): implement AppSwitch component with Material Design 3 compliance

mrezan12 opened the pull request:
## 📄 Summary

Implement the [AppSwitch](cci:2://file:///home/mrcan/Documents/projects/hexaMobileShare/packages/core_kit/lib/widgets/inputs/app_switch.dart:95:0-276:1) component with Material Design 3 compliance for the `core_kit` package.

This PR adds a fully functional toggle switch widget with:
- Label, subtitle, and icon support
- On/off and disabled states
- Custom color options
- Comprehensive accessibility features (48x48dp touch targets, semantic labels)
- 26 widget tests with 100% pass rate
- 7 Widgetbook stories showcasing all features

Closes #17

---

## 🧩 Affected Module(s)

- [x] Packages (packages/*)
- [x] Widgetbook Kit (widgetbook_kit/)
- [ ] Documentation (docs/)
- [ ] CI / Infra (.github/)

---

## ✅ Checklist

- [✅] My branch name follows format: `<type>/<short-description>` (e.g., feat/login-screen)
- [✅] My PR title starts with one of the approved types listed above
- [✅] My Dart code is formatted (`dart format .` or `flutter format .`)
- [✅] I ran static analysis (`flutter analyze`) and resolved warnings
- [✅] I ran tests successfully (`flutter test`)
- [✅] I updated / checked pubspec.yaml and pubspec.lock for dependency changes
- [✅] For UI changes, I added screenshots and/or updated golden tests (if applicable)
- [✅] I linked related issues using keywords like `Closes #42`
- [✅] I ensured this PR has no unrelated changes
- [✅] This PR is ready for review and does not include unfinished work

---

## 🔗 Related Issues

Closes #17

---

## 💬 Additional Notes

### Files Changed:
- [packages/core_kit/lib/widgets/inputs/app_switch.dart](cci:7://file:///home/mrcan/Documents/projects/hexaMobileShare/packages/core_kit/lib/widgets/inputs/app_switch.dart:0:0-0:0) - Widget implementation
- [packages/core_kit/test/widgets/inputs/app_switch_test.dart](cci:7://file:///home/mrcan/Documents/projects/hexaMobileShare/packages/core_kit/test/widgets/inputs/app_switch_test.dart:0:0-0:0) - 26 widget tests
- [widgetbook_kit/lib/stories/core_kit/widgets/inputs/app_switch_stories.dart](cci:7://file:///home/mrcan/Documents/projects/hexaMobileShare/widgetbook_kit/lib/stories/core_kit/widgets/inputs/app_switch_stories.dart:0:0-0:0) - 7 stories
- [widgetbook_kit/lib/app/widgetbook_app.directories.g.dart](cci:7://file:///home/mrcan/Documents/projects/hexaMobileShare/widgetbook_kit/lib/app/widgetbook_app.directories.g.dart:0:0-0:0) - Generated

### Test Results:
- ✅ 26/26 tests passed (100% pass rate)
- ✅ `flutter analyze` - No issues found
- ✅ `reuse lint` - REUSE compliant
- ✅ All CI checks passed

### Widgetbook Stories:
1. Default
2. All States
3. With Labels
4. With Icons
5. Settings List
6. Theme Variations
7. Real-World Examples

mrezan12 committed:
feat(core-kit): implement AppSwitch component with Material Design 3 compliance

- Implement AppSwitch widget as StatelessWidget with comprehensive properties
- Add support for label, subtitle, and icon customization
- Implement Material Design 3 color tokens and specifications
- Support both on/off states and disabled states
- Add accessibility features (48x48dp touch targets, semantic labels)
- Create 26 comprehensive widget tests with 100% pass rate
- Develop 7 Widgetbook stories showcasing all features and use cases
- Ensure REUSE compliance with proper SPDX headers
- Fix deprecated Switch API (activeColor → thumbColor/trackColor)

Closes #17

husamettinarabaci reviewed (changes requested):
## 🔍 Manual Testing Results

I tested the AppSwitch component using `pnpm storybook` and discovered a critical issue with the Widgetbook stories implementation.

### ❌ Issue: Non-Interactive Stories

**Problem:** While all switches render correctly visually, **none of them respond to user interaction** (taps/clicks) in Widgetbook.

**Root Cause:** All 7 Widgetbook stories use static `value` parameters and empty `onChanged` callbacks without state management:

```dart
// Current implementation (app_switch_stories.dart)
AppSwitch(
  value: false,  // ❌ Hard-coded static value
  label: 'Toggle switch',
  onChanged: (value) {},  // ❌ Empty callback - no state update
)
```

This means users **cannot test the interactive behavior** of switches in Widgetbook, which defeats the purpose of having interactive documentation.

---

## 📊 Analysis of All Stories

| Story | Interactive? | State Management | Status |
|-------|--------------|------------------|--------|
| Default | ❌ No | None | Needs fix |
| All States | ⚠️ Static demo | None | OK (comparison view) |
| With Labels | ❌ No | None | Needs fix |
| With Icons | ❌ No | None | Needs fix |
| Settings List | ❌ No | None | **Critical - should be fully interactive** |
| Theme Variations | ⚠️ Static demo | None | OK (color comparison) |
| Real-World Examples | ❌ No | None | Needs fix |

**Required fixes:** 5 out of 7 stories need to be made interactive.

---

## 💡 Solution: Use StatefulWidget for Interactive Stories

Stories that demonstrate user interaction **must** use `StatefulWidget` to manage state:

### Example Fix:

```dart
// ✅ CORRECT: Interactive story with state management
@widgetbook.UseCase(name: 'Default', type: AppSwitch)
Widget appSwitchDefault(BuildContext context) {
  return _InteractiveSwitchDemo();
}

class _InteractiveSwitchDemo extends StatefulWidget {
  @override
  State<_InteractiveSwitchDemo> createState() => _InteractiveSwitchDemoState();
}

class _InteractiveSwitchDemoState extends State<_InteractiveSwitchDemo> {
  bool _value = false;

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(16.0),
      child: AppSwitch(
        value: _value,  // ✅ Value comes from state
        label: 'Toggle switch',
        onChanged: (value) {
          setState(() => _value = value);  // ✅ Updates state on change
        },
      ),
    );
  }
}
```

---

## 🎯 Stories That Need Updates

### 1. **Default** (Critical)
- Should be fully interactive
- Users should be able to toggle on/off

### 2. **With Labels** (High Priority)
- At least 2-3 examples should be interactive
- Demonstrates that labels are clickable

### 3. **With Icons** (High Priority)
- All 4 examples (WiFi, Bluetooth, Airplane, Location) should be interactive
- Shows realistic behavior

### 4. **Settings List** (Critical - Most Important!)
- This is the showcase story - must be fully interactive
- All 11 switches should toggle independently
- This demonstrates real-world usage
- Currently shows a realistic UI but none of the switches work!

### 5. **Real-World Examples** (High Priority)
- All 4 examples should be interactive
- Dark mode, auto-save, cloud backup should toggle
- Analytics switch should remain disabled (correct as-is)

---

## 📝 Implementation Pattern

For stories with multiple switches, use a single StatefulWidget with multiple state variables:

```dart
@widgetbook.UseCase(name: 'Settings List', type: AppSwitch)
Widget appSwitchSettingsList(BuildContext context) {
  return _SettingsListDemo();
}

class _SettingsListDemo extends StatefulWidget {
  @override
  State<_SettingsListDemo> createState() => _SettingsListDemoState();
}

class _SettingsListDemoState extends State<_SettingsListDemo> {
  // State for each switch
  bool _wifi = true;
  bool _bluetooth = true;
  bool _airplane = false;
  bool _notifications = true;
  bool _vibrate = false;
  bool _location = true;
  bool _camera = false;
  bool _mic = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Settings')),
      body: ListView(
        children: [
          // Network & Connectivity
          AppSwitch(
            value: _wifi,
            icon: Icons.wifi,
            label: 'Wi-Fi',
            onChanged: (value) => setState(() => _wifi = value),
          ),
          AppSwitch(
            value: _bluetooth,
            icon: Icons.bluetooth,
            label: 'Bluetooth',
            onChanged: (value) => setState(() => _bluetooth = value),
          ),
          // ... etc
        ],
      ),
    );
  }
}
```

---

## 🔬 Testing Process Importance

### Why `pnpm storybook` Testing is Critical

This issue highlights why **manual testing in Widgetbook is essential** before PR approval:

1. **Visual Tests ≠ Functional Tests**
   - Static stories can look perfect but be non-functional
   - CI tests don't catch non-interactive Widgetbook stories

2. **Widgetbook is Interactive Documentation**
   - Developers use Widgetbook to **understand component behavior**
   - Non-interactive stories provide incomplete documentation
   - Users can't experiment with the component

3. **Quality Assurance Gate**
   - Running `pnpm storybook` should be part of PR checklist
   - Test all interactive elements before marking PR as "ready for review"

4. **Real-World Validation**
   - Settings List story looks production-ready but is completely static
   - Without testing, we can't verify actual usability

---

## ✅ Acceptance Criteria for Fix

Before this PR can be merged:

- [ ] Default story is interactive (switch toggles on click)
- [ ] With Labels story has at least 2 interactive examples
- [ ] With Icons story - all 4 switches are interactive
- [ ] **Settings List story - all switches work independently** (most critical)
- [ ] Real-World Examples - 3 active switches work, 1 disabled (Analytics)
- [ ] All States and Theme Variations can remain static (they're comparison views)
- [ ] Test in Widgetbook: `pnpm storybook` - verify all switches respond to clicks
- [ ] Re-generate Widgetbook directories: `pnpm generate:widgetbook`

---

## 📚 Reference: Correct Implementation Example

**PR #167 (AppFab)** did this correctly:

- `_SwitchableDemo` uses StatefulWidget (lines 146-176)
- `_LoadingFab` uses StatefulWidget with animation (lines 206-246)
- Interactive playground demonstrates real behavior

**Suggested approach:** Follow the same pattern used in AppFab stories.

---

## 🎯 Impact Assessment

**Severity:** High  
**Reason:** The component implementation is excellent (26 tests, great docs), but Widgetbook stories don't demonstrate the core functionality (toggling). This significantly reduces the value of the design system documentation.

**User Impact:**
- Developers using Widgetbook can't test switch behavior
- Cannot verify that switches work before integrating into their app
- Settings List story looks professional but is misleading (appears interactive but isn't)

---

## 💬 Additional Notes

**Great work on:**
- ✅ Component implementation (273 lines, clean code)
- ✅ 26 comprehensive tests (100% pass rate)
- ✅ Excellent documentation (best of all reviewed PRs)
- ✅ Material Design 3 compliance
- ✅ Story visual design (Settings List looks production-ready)

**This is a high-quality PR** - it just needs the Widgetbook stories to be made interactive to match the quality of the component implementation.

Please update the stories and I'll be happy to approve! Let me know if you need any clarification on the implementation pattern.

mrezan12 committed:
fix(widgetbook): make AppSwitch stories interactive

- Convert 5 stories to use StatefulWidget for state management
- Default, With Labels, With Icons, Settings List, Real-World Examples now interactive
- All States and Theme Variations remain static (comparison views)
- Settings List: 8 active switches + 1 disabled (Do not disturb)
- Real-World Examples: 3 active switches + 1 disabled (Analytics)
- Addresses PR #165 review feedback

mrezan12 committed:
Merge origin/develop into feat/app-switch-component

mrezan12 committed:
fix(widgetbook): make Theme Variations story interactive

- Convert Theme Variations from static to StatefulWidget
- All 5 switches (default on/off, green, orange, blue) now toggle
- Merged develop into feat/app-switch-component

husamettinarabaci reviewed (dismissed):
LGTM

husamettinarabaci committed:
Merge branch 'develop' into feat/app-switch-component
